tests/ignite/metrics/test_mean_absolute_error.py

```diff
@@ -17,20 +18,47 @@ def test_zero_div():
 
 
 def test_compute():
-    mae = MeanAbsoluteError()
 
-    y_pred = torch.Tensor([[2.0], [-2.0]])
-    y = torch.zeros(2)
-    mae.update((y_pred, y))
-    assert isinstance(mae.compute(), float)
-    assert mae.compute() == 2.0
+    mae = MeanAbsoluteError()
 
-    mae.reset()
-    y_pred = torch.Tensor([[3.0], [-3.0]])
-    y = torch.zeros(2)
-    mae.update((y_pred, y))
-    assert isinstance(mae.compute(), float)
-    assert mae.compute() == 3.0
+    def _test(y_pred, y, batch_size):
+        mae.reset()
+        mae.update((y_pred, y, batch_size))
+
+        np_y = y.numpy()
+        np_y_pred = y_pred.numpy()
+
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                mae.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+
+        np_res = (np.abs(np_y_pred - np_y)).sum() / np_y.shape[0]
+        assert isinstance(mae.compute(), float)
+        assert mae.compute() == np_res
+
+    def get_test_cases():
```

Comment:
@KickItLikeShika @vfdev-5 

wouldn’t we use `pytest.parametrize` here ?

Reply:
@sdesrozis we discussed that in some of previous PRs and as we run `_test` multiple times, we prefer to generate different data points. Using parametrize it would be random data but the same for `for _ in range(10)` loop. Valid question is whether we would need that loop...

Reply:
Sorry for the delay @sdesrozis, and thanks for explanation @vfdev-5. Is everything fine now?

Reply:
> Valid question is whether we really need that loop...

We need the loop if random values can generate different behaviour. I think that is not useful here.